sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java

```diff
@@ -374,159 +368,132 @@ public void deleteContact(String contactUuid) {
 
 		cq.orderBy(cb.desc(contact.get(Contact.REPORT_DATE_TIME)));
 
-		List<ContactExportDto> resultList = em.createQuery(cq).setFirstResult(first).setMaxResults(max).getResultList();
-
-		for (ContactExportDto exportDto : resultList) {
-			exportDto.setNumberOfVisits(visitService.getVisitCountByContactId(exportDto.getPersonId(),
-					exportDto.getLastContactDate(), exportDto.getReportDate(), exportDto.getFollowUpUntil(),
-					exportDto.getInternalDisease()));
-			Visit lastCooperativeVisit = visitService.getLastVisitByContactId(exportDto.getPersonId(),
-					exportDto.getLastContactDate(), exportDto.getReportDate(), exportDto.getFollowUpUntil(),
-					exportDto.getInternalDisease(), VisitStatus.COOPERATIVE);
-			if (lastCooperativeVisit != null) {
-				exportDto.setLastCooperativeVisitSymptomatic(lastCooperativeVisit.getSymptoms().getSymptomatic() ?
-						YesNoUnknown.YES : YesNoUnknown.NO);
-				exportDto.setLastCooperativeVisitDate(lastCooperativeVisit.getVisitDateTime());
-				exportDto.setLastCooperativeVisitSymptoms(lastCooperativeVisit.getSymptoms().toHumanString(true, userLanguage));
+		List<ContactExportDto> exportContacts = em.createQuery(cq).setFirstResult(first).setMaxResults(max).getResultList();
+
+		if (!exportContacts.isEmpty()) {
+			List<Long> exportContactIds = exportContacts.stream().map(e -> e.getId()).collect(Collectors.toList());
+
+			CriteriaQuery<VisitSummaryExportDetails> visitsSq = cb.createQuery(VisitSummaryExportDetails.class);
+			Root<Contact> visitsSqRoot = visitsSq.from(Contact.class);
+			Join<Contact, Visit> visitsJoin = visitsSqRoot.join(Contact.VISITS, JoinType.LEFT);
+			Join<Visit, Symptoms> visitSymptomsJoin = visitsJoin.join(Visit.SYMPTOMS, JoinType.LEFT);
+
+			visitsSq.where(ContactService.and(cb,
+					contact.get(AbstractDomainObject.ID).in(exportContactIds),
+					cb.isNotEmpty(visitsSqRoot.get(Contact.VISITS)))
+					);
+			visitsSq.multiselect(
+					visitsSqRoot.get(AbstractDomainObject.ID),
+					visitsJoin.get(Visit.VISIT_DATE_TIME),
+					visitsJoin.get(Visit.VISIT_STATUS),
+					visitSymptomsJoin
+					);
+
+			List<VisitSummaryExportDetails> contactVisits = em.createQuery(visitsSq).getResultList();
```

Comment:
Add comment that explains that this is not perfect, but a one-querry solution won't work because ...